activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/test/java/org/activiti/cloud/services/identity/keycloak/PublicKeyValidationCheckIT.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2017-2020 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.activiti.cloud.services.identity.keycloak;
+
+import static io.smallrye.common.constraint.Assert.assertFalse;
+import static io.smallrye.common.constraint.Assert.assertTrue;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.activiti.api.runtime.shared.security.SecurityContextTokenProvider;
+import org.activiti.cloud.services.identity.keycloak.validator.PublicKeyValidationCheck;
+import org.activiti.cloud.services.test.containers.KeycloakContainerApplicationInitializer;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.test.context.ContextConfiguration;
+
+@SpringBootTest(
+    classes = {KeycloakClientApplication.class},
+    properties = {
+        "keycloak.realm=activiti",
+        "keycloak.use-resource-role-mappings=false"}
+)
+@ContextConfiguration(initializers = {KeycloakContainerApplicationInitializer.class})
+class PublicKeyValidationCheckIT {
+
+    @Autowired
+    private PublicKeyValidationCheck publicKeyValidationCheck;
+    @Value("${keycloak.auth-server-url}") String authServerUrl;
+    @Value("${keycloak.realm}") String realm;
+
+    @Autowired
+    ObjectMapper objectMapper;
+
+    @Autowired
+    SecurityContextTokenProvider securityContextTokenProvider;
+
+    @Test
+    void should_tokenBeValid() {
+        Jwt accessToken = Jwt.withTokenValue(securityContextTokenProvider.getCurrentToken().get())
+            .header("a", "b")
+            .claim("a", "b")
```

Comment:
What are the meaning of this fields values?

Reply:
they are mandatory for the builder, but in the end they aren't used.